src/utilities/filetypes/AlfalfaJSON.cpp

```diff
@@ -68,45 +75,40 @@ namespace alfalfa {
 
     Json::Value AlfalfaJSON_Impl::toJSON() const {
       Json::Value root;
-      for (AlfalfaPoint point : m_points) {
+      for (const auto& point : m_points) {
         // No guard here as the toJSON call will throw an exception if the id does not exist.
         root[point.id().get()] = point.toJSON();
       }
       return root;
     }
   }  // namespace detail
 
-  AlfalfaJSON::AlfalfaJSON() : m_impl(std::shared_ptr<detail::AlfalfaJSON_Impl>(new detail::AlfalfaJSON_Impl())) {}
+  AlfalfaJSON::AlfalfaJSON() : m_impl(std::make_shared<detail::AlfalfaJSON_Impl>()) {}
 
-  AlfalfaJSON::AlfalfaJSON(const std::string& s) : m_impl(std::shared_ptr<detail::AlfalfaJSON_Impl>(new detail::AlfalfaJSON_Impl(s))) {}
+  AlfalfaJSON::AlfalfaJSON(const std::string& s) : m_impl(std::make_shared<detail::AlfalfaJSON_Impl>(s)) {}
 
-  AlfalfaJSON::AlfalfaJSON(const openstudio::path& p) : m_impl(std::shared_ptr<detail::AlfalfaJSON_Impl>(new detail::AlfalfaJSON_Impl(p))) {}
+  AlfalfaJSON::AlfalfaJSON(const openstudio::path& p) : m_impl(std::make_shared<detail::AlfalfaJSON_Impl>(p)) {}
 
   boost::optional<AlfalfaPoint> AlfalfaJSON::exposeConstant(float value, const std::string& display_name) {
-    AlfalfaConstant component(value);
-    return exposeFromComponent(component, display_name);
+    return exposeFromComponent(AlfalfaConstant{value}, display_name);
```

Comment:
I'm not familiar with this method of constructing, this is just a way to construct the class inline in the stack?

Reply:
This is by definition a temporary (because it does not have a name), so it's an R-value, and I know that there will be no copies.

In this specific case, this is unnecessary, because the `exposeFromComponent` actually takes a const-ref, so no copy or move will happen, but that may change... **It didn't hinder readability much here**, so I just went with it.

This is heavily related to the pass-by-value and move idiom.

Here is something to play for you, switch the defines on line 41 and 42: https://gcc.godbolt.org/z/911xK1bT6

You'll see that if exposeFromComponent takes by-value:
* If you NAME your variable, the Ctor is called, then the Copy Ctor
* If you use a temporary (like I did here), the variable is actually constructed directly in the exposeFromComponent scope and you save a copy.